src/bidiMapper/modules/network/NetworkStorage.ts

```diff
@@ -327,7 +475,11 @@ export class NetworkStorage {
     this.#requests.set(request.id, request);
   }
 
-  deleteRequest(id: Network.Request) {
+  disposeRequest(id: Network.Request) {
+    if (this.#requestCollectors.get(id)?.size ?? 0 > 0) {
+      // Keep request is collected. Keep it, as it's data can be accessed later.
+      return;
+    }
```

Comment:
let's add a TODO here to call CDP once we have a command to remove a specific request